src/injected/element-based-view-model-creator.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { CardSelectionViewData } from 'common/get-card-selection-view-data';
+import { CardSelectionStoreData } from 'common/types/store-data/card-selection-store-data';
+import { UnifiedResult, UnifiedRule } from 'common/types/store-data/unified-data-interface';
+import { AssessmentVisualizationInstance } from 'injected/frameCommunicators/html-element-axe-results-helper';
+import { GetDecoratedAxeNodeCallback } from 'injected/get-decorated-axe-node';
+import { find, includes } from 'lodash';
+import { DictionaryStringTo } from 'types/common-types';
+
+export interface CheckData {
+    // tslint:disable-next-line: no-reserved-keywords
+    any: FormattedCheckResult[];
+    none: FormattedCheckResult[];
+    all: FormattedCheckResult[];
+}
+
+export type GetElementBasedViewModelCallback = (
+    rules: UnifiedRule[],
+    results: UnifiedResult[],
+    cardSelectionData: CardSelectionStoreData,
+) => DictionaryStringTo<AssessmentVisualizationInstance>;
+
+export type GetHighlightedResultInstanceIdsCallback = (
+    cardSelectionData: CardSelectionStoreData,
+) => Pick<CardSelectionViewData, 'highlightedResultUids'>;
+
+export class ElementBasedViewModelCreator {
+    constructor(
+        private getDecoratedAxeNode: GetDecoratedAxeNodeCallback,
+        private getHighlightedResultInstanceIds: GetHighlightedResultInstanceIdsCallback,
+    ) {}
+
+    public getElementBasedViewModel: GetElementBasedViewModelCallback = (
+        rules: UnifiedRule[],
+        results: UnifiedResult[],
+        cardSelectionData: CardSelectionStoreData,
+    ) => {
+        const stringToResult: DictionaryStringTo<AssessmentVisualizationInstance> = {};
```

Comment:
I would love it if you renamed this variable to resultsDictionary or something like that. As is, the variable sounds like a function, which in JavaScript can be quite confusing.